asiana21 DOCSUP-194 (#86)
BayoNet/ClickHouse: docs/en/query_language/functions/type_conversion_functions.md

└───────────────────────────┴──────────────────────────────┘
```

## parseDateTimeBestEffort {#parsedatetimebesteffort}

Converts a [String](../../data_types/string.md) to [DateTime](../../data_types/datetime.md#data_type-datetime) data type.

> The function parses ISO 8601, RFC 1123 and more complex date formats.

Reviewer: ```suggestion
The function parses ISO 8601, RFC 1123 and some other date and time formats.
```

**Syntax**

```sql
parseDateTimeBestEffort(s[, time_zone]);
```

**Parameters**

> - `s` — A string containing a date and time to convert. [String](../../data_types/string.md).  

Reviewer: ```suggestion
- `s` — String containing a date and time to convert. [String](../../data_types/string.md).  
```

> - `time_zone` — Optional second argument. Parse `s` according to the time zone. [String](../../data_types/string.md).

Reviewer: ```suggestion
- `time_zone` — Time zone. The function parses `s` according to the time zone. [String](../../data_types/string.md).
```

**Supported non-standart formats**

> - A string containing 9..10 digits: `NNNNNNNNNN`.

Reviewer: ```suggestion
- A string containing 9..10 digit unixtimestamp.
```

└───────────────────────────┴──────────────────────────────┘
```

## parseDateTimeBestEffort {#parsedatetimebesteffort}

> Converts a string representation of a date and time value to its DateTime equivalent.

Reviewer: DateTime is a data type for storage. It is not an equivalent. Change the sentence and add a link to the data type.

> Different from toDate and toDateTime, parseDateTimeBestEffort can progress more complex date format.

Reviewer: I think we don't need this sentence. Refer these functions from **See Also**. Refer parseDateTimeBestEffort from the functions with the words about a wider range of input formats available.

Also, we always emphasize keywords with backticks (`toDate`) or with link([toDate](#)

**Syntax**
```sql
parseDateTimeBestEffort(s[, time_zone]);
```

**Parameters**

> - `s` — A string containing a date to convert. [String](../../data_types/string.md).  

Reviewer: Why only date?

>     The string to be parsed can take any of the following forms:

Reviewer: I think we need a separate section for input formats description. Let's try to place it after Parameters and see how it looks.

    - A string with a time but no date component: `hh:mm:ss`, `hh:mm`, `hh` - if already have day of month.
    - A string conforms to the RFC 1123 time format: `Tue, 02 Sep 2018 20:01:00 GMT`, etc.
    - A string that includes time zone information and conforms to ISO 8601: `2020-12-12T17:36:00.0000000Z`, etc.
    - A string that includes the date and time along with time zone offset information: `2020-12-12 17:36:00 -5:00`, etc.
 
- `time_zone` — Optional second argument. Parse `s` according to the time zone. [String](../../data_types/string.md).

**Returned value**

- Data structure equivalent to the date specified in `s`.

Type: `DateTime`.

**Example**

Query:

```sql
SELECT parseDateTimeBestEffort('12/12/2020 12:12:57')
AS parseDateTimeBestEffort;
```

Result:

```text
┌─parseDateTimeBestEffort─┐
│     2020-12-12 12:12:57 │
└─────────────────────────┘
```

Query:

```sql
SELECT parseDateTimeBestEffort('Sat, 18 Aug 2018 07:22:16 GMT', 'Europe/Moscow') 
AS parseDateTimeBestEffort
```

Result:

```text
┌─parseDateTimeBestEffort─┐
│     2018-08-18 10:22:16 │
└─────────────────────────┘
```

**See Also**

> - [Complex Date Format](https://xkcd.com/1179/)

Reviewer: ```suggestion
- [ISO 8601 announcement by @xkcd](https://xkcd.com/1179/)
```

Add links to other date and time conversion functions and to functions for working with dates and times.
